pkg/operator/baremetal_config.go

```diff
@@ -132,10 +132,15 @@ func getIronicInspectorEndpoint(baremetalConfig BaremetalProvisioningConfig) *st
 }
 
 func getProvisioningDHCPRange(baremetalConfig BaremetalProvisioningConfig) *string {
+	// When the DHCP server is external, it is OK for the DHCP range in the CR
+	// to be empty.
 	if baremetalConfig.ProvisioningDHCPRange != "" {
 		return &(baremetalConfig.ProvisioningDHCPRange)
+	} else if baremetalConfig.ProvisioningDHCPExternal {
+		return &(baremetalConfig.ProvisioningDHCPRange)
+	} else {
+		return nil
 	}
```

Comment:
Nit: This else is redundant since all other blocks return (this is a go idiom that a linter would normally flag up) 
```suggestion
	} 
	return nil
```